### PR TITLE
Troubleshooting, caching, node setup guide updates

### PR DESCRIPTION
**Pull request by johannakate**
… retention how to contact us pages

### CODE REVIEW

**Comment by julianeon** [comment]
At one point you show the log output that indicates the Engine proxy is healthy. Applying Jesse's comments (in a different context) here, it would be good to reduce the size of the output so that it takes up less space with only the necessary part shown. The rest of the output can be implied by ... or redacted.

File: source/troubleshooting.md [hunk 1]
```diff
@@ -118,19 +118,13 @@ If using the sidecar deployment configuration - check that you integrated the En
 
 <h2 id="">Troubleshooting FAQs</h2>
 
-**I'm getting an error saying “The query failed!”, how do I fix it?**
+**I'm receiving an error. What does this mean?**
 
-This may mean you need to upgrade an NPM package. Check that your package versions are all up-to-date. This also may mean a variety of other things. When this error is paired with a 503 error, the query did not receive an expected response.
+| Error message | Issue description and resolution steps |
+| 503 error - unable to communicate with backend | This likely means that the origin GraphQL server did not respond to the proxied request. To resolve this, verify the Origin specification in the Engine configuration matches the corresponding GraphQL server location |
```
Comment (abernix) [inline source/troubleshooting.md]: Let's change Origin to `origin` (in backticks) to match the Engine configuration format.

File: source/troubleshooting.md [hunk 2]
```diff
@@ -118,19 +118,13 @@ If using the sidecar deployment configuration - check that you integrated the En
...
+| 403 error - [location unavailable] | Received unsupported Content-Type from origin GraphQL server. This often occurs when a GraphQL server throws an error during resolver execution and sends that error as `text/plain` instead of embedding it within a GraphQL error. This could also be caused by an inaccurate Content-Type header. |
```
Comment (abernix) [inline source/troubleshooting.md]: I would vote for both of these "Content-type" references to be `Content-type` (backticked).

File: source/troubleshooting.md [hunk 3]
```diff
@@ -118,19 +118,13 @@ If using the sidecar deployment configuration - check that you integrated the En
 
 <h2 id="">Troubleshooting FAQs</h2>
 
-**I'm getting an error saying “The query failed!”, how do I fix it?**
+**I'm receiving an error. What does this mean?**
 
-This may mean you need to upgrade an NPM package. Check that your package versions are all up-to-date. This also may mean a variety of other things. When this error is paired with a 503 error, the query did not receive an expected response.
+| Error message | Issue description and resolution steps |
```
Comment (abernix) [inline source/troubleshooting.md]: I'm not positive a table will be the best format for this amount of text, but will wait and see with the formatting working.  You can check the Netlify preview on the Pull-Request after submitting it, but please reach out to me if you're having problems testing it locally!

File: source/troubleshooting.md [hunk 4]
```diff
@@ -140,13 +134,61 @@ Third, we currently support a certain range of language-specific GraphQL servers
...
+   { address: '127.0.0.1:64201',
+     level: 'info',
+     msg: 'Started HTTP server.',
+     time: '2018-03-01T12:33:35-08:00' } }
+{ proxy:
+   { configured: 3,
+     current: 0,
+     level: 'debug',
+     msg: 'Synchronizing stores.',
+     time: '2018-03-01T12:33:35-08:00' } }
```
Comment (abernix) [inline source/troubleshooting.md]: While we encourage cache configuration, some of these "store" related messages won't appear unless they are enabled, so we should probably limit this to the messages which show up for all users.

That will also allow this section to be more brief, and I think we can find a way to form this information in a way that it still gets across the important deets.

File: source/data-retention.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
```
[the remark this answers is not in the document]
Comment (julianeon) [reply source/data-retention.md]: Thanks for mentioning this. Please do add comments talking about style and best practices for docs PR's, as in my case, I can personally say I don't know them. I was wondering if it's better to have a big 'all-in-one' PR that covers multiple things, vs. multiple PR's each w/their own theme, and this answers it.

File: source/alerts.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+---
+title: Alerts
+sidebar_title: Alerts
+description: Create triggers for GraphQL errors or p95/99 latency thresholds and respond not only faster, but smarter.
+---
+<h2 id="engine-alerts" title=">Engine Alerts</h2>
+
+Performance metrics, such as service time and error rates, are indicators of your customer's experiences: these two metrics alone can make or break a technology company. Engine is an reliable solution for any team looking to enable alerts when user experience on your application begins to decline and you need a call to action.
+
+<h2 id="alert-channels" title="Enable Alerts">Enable Alerts</h2>
```
[the remark this answers is not in the document]
Comment (julianeon) [reply source/alerts.md]: Spelling error - an reliable solution should be a reliable solution.

Grammatical - customer's experiences probably shouldn't be singular

To sidestep that, I'd change it to "indicators of the customer experience"

More generally, for style:

Performance metrics, such as service time and error rates, are indicators of the customer experience: these metrics alone can make or break a technology company. Engine is a reliable solution for any team looking to enable alerts when the user experience on your application declines and you need a call to action.

File: source/data-retention.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (johannakate) [reply source/data-retention.md]: +1 on Julian's comment, these comments are awesome thanks @abernix!

[the remark this answers is not in the document]
Comment (abernix) [reply source/data-retention.md]: @julianeon We can incrementally add things faster in this way, rather than having an entire PR blocked because of an unrelated section.  It's also more likely that one of the many "reviewers" might have a few seconds for a smaller PR, rather than a PR that changes every section.

I do think, however, that similar changes (and especially repetitive changes, like changing the same thing in 100 places across 10 docs) should be grouped together.  I think best practice is flexible here, but that's my thought.

File: source/alerts.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (stubailo) [inline source/alerts.md]: Can we move this to a separate PR? I don't think we know that we want to advertise this feature here.

File: source/data-retention.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (stubailo) [reply source/data-retention.md]: Agreed - can we move the stuff about upsell features to separate PRs, one for data retention and one for alerts? I don't think we can merge those yet.

**Comment by glasser** [comment]
This conflicts with the changes Sashko and I made last week as part of 1.0. Let me know if you've updated the PR and want to go forward with it.

**Comment by stubailo** [comment]
Closing for now since a lot of these things are re-done. I think splitting the PR into smaller parts will make it easier to merge faster.
